app/services/ucas_matching/matching_data_file.rb

```diff
@@ -0,0 +1,41 @@
+require 'csv'
+# require 'archive/zip'
+
+module UCASMatching
+  class MatchingDataFile
+    # @returns filename
+    def create_file
+      filename_prefix = "#{Rails.root}/tmp/dfe_apply_itt_applications_#{Time.zone.now.iso8601}-#{SecureRandom.hex}"
+      csv_filename = "#{filename_prefix}.csv"
+      zip_filename = "#{filename_prefix}.zip"
+
+      File.open(csv_filename, 'w') do |f|
+        f.write(csv_as_string)
```

Comment:
Do we need the intermediate CSV file — could we pass it to `Archive::Zip` as a `StringIO`, maybe?

Reply:
Maybe that wouldn't work — I guess the ZIP needs to take named files on disk or what would it unzip to?

Reply:
Yeah, I think `archive-zip` can't do it without the intermediate file.